code/modules/antagonists/contractor/contract.dm

```diff
@@ -237,15 +237,15 @@
 
 		do_sparks(8, FALSE, target)
 		target.visible_message(span_notice("[target] vanishes..."))
-
-		for(var/obj/item/target_item as anything in target)
-			if(ishuman(target))
-				var/mob/living/carbon/human/human_target = target
-				if(target_item == human_target.w_uniform)
-					continue //So all they're left with are shoes and uniform.
-				if(target_item == human_target.shoes)
-					continue
-			target.dropItemToGround(target_item)
+		if(istype(target))
+			for(var/obj/item/target_item as anything in target.get_all_worn_items())
+				if(ishuman(target))
+					var/mob/living/carbon/human/human_target = target
+					if(target_item == human_target.w_uniform)
+						continue //So all they're left with are shoes and uniform.
+					if(target_item == human_target.shoes)
+						continue
+				target.dropItemToGround(target_item)
```

Comment:
replace `istype()` with `ishuman()` and move the var assignment to out of the loop. The only carbons that exist are Xenos so it's fine.

Reply:
done